Note container runtime requirement for image volume subPath (#56606)
kubernetes/website: content/en/docs/tasks/configure-pod-container/image-volumes.md

[`subPathExpr`](/docs/concepts/storage/volumes/#using-subpath-expanded-environment)
from Kubernetes v1.33 when using the image volume feature.

> {{< note >}}

Reviewer: Add a third-party content disclaimer before naming specific runtimes.

```suggestion
{{% thirdparty-content %}}

{{< note >}}
```

> `subPath` and `subPathExpr` for image volumes are applied by the container runtime,

Reviewer: ```suggestion
The container runtime handles the `subPath` and `subPathExpr` fields for image   
  volumes, 
```

so a recent enough Kubernetes version is not sufficient on its own. This support is
available since containerd v2.2 and CRI-O v1.33.

> If your container runtime does not support it, the field is ignored and the whole

Reviewer: ```suggestion
If your container runtime does not support this, it ignores the field and mounts the whole
```
